query_server/query/src/metadata.rs

```diff
@@ -166,6 +176,183 @@ impl MetaData for LocalCatalogMeta {
         Ok(())
     }
 
+    fn show_database(&self) -> Result<Output> {
+        let dbs = self.engine.list_databases().unwrap();
+
+        let schema = Arc::new(Schema::new(vec![Field::new(
+            "Database",
+            DataType::Utf8,
+            false,
+        )]));
+
+        let batch = RecordBatch::try_new(schema, vec![Arc::new(StringArray::from(dbs))]).unwrap();
+
+        let batches = vec![Arc::new(batch)];
+
+        Ok(Output::StreamData(stream_from_batches(batches)))
+    }
+
+    fn show_table(&self, name: &str) -> Result<Output> {
+        let mut database_name = name;
+        if database_name.is_empty() {
+            database_name = self.database_name.as_str()
+        }
+
+        match self.catalog.schema(database_name) {
+            None => Err(MetadataError::DatabaseNotExists {
+                database_name: database_name.to_string(),
+            }),
+            Some(_db_cfgs) => {
+                let schema = Arc::new(Schema::new(vec![Field::new(
+                    "Table",
+                    DataType::Utf8,
+                    false,
+                )]));
+
+                let tables = self.engine.list_tables(database_name).unwrap();
```

Comment:
May panic if an error is returned. In principle, all places that return `Option` or `Result` need to handle.

Reply:
nice hint